src/Swarm/TUI/Controller.hs

```diff
@@ -398,8 +403,14 @@ handleModalEvent = \case
     Brick.zoom (uiState . uiModal . _Just . modalDialog) (handleDialogEvent ev)
     modal <- preuse $ uiState . uiModal . _Just . modalType
     case modal of
+      Just GoalModal -> do
+        lw <- use $ uiState . uiGoal . listWidget
+        newList <- refreshList lw
+        uiState . uiGoal . listWidget .= newList
       Just _ -> handleInfoPanelEvent modalScroll (VtyEvent ev)
       _ -> return ()
+   where
+    refreshList lw = nestEventM' lw $ BL.handleListEvent ev
```

Comment:
It would be nice to skip over the headers when moving around the goal list with arrow keys, similar to what we do in the inventory.  We already have a function `handleListEventWithSeparators` to handle this.

https://github.com/swarm-game/swarm/blob/7fd2a1cc1a164af25292e7b80ff94e250034bf83/src/Swarm/TUI/Controller.hs#L1100

https://github.com/swarm-game/swarm/blob/181e3c43e548ec1f184f75ebf1e5a272b2c0b572/src/Swarm/TUI/List.hs#L20-L29

Reply:
Works great!